src/codewhisperer/commands/basicCommands.ts

```diff
@@ -120,37 +121,43 @@ export const showSsoSignIn = Commands.declare('aws.codeWhisperer.sso', () => asy
-// It can optionally set a customization too.
+// It can optionally set a customization too based on given values to match on
 export const connectWithCustomization = Commands.declare(
     'aws.codeWhisperer.connect',
-    () =>
-        async (
-            startUrl?: string,
-            region?: string,
-            customizationArn?: string,
-            customizationName?: string,
-            customizationDescription?: string
-        ) => {
-            // This command supports two sets of arguments:
-            //  * startUrl and region. If both arguments are provided they will be used, otherwise
-            //    the command prompts for them interactively.
-            //  * customization{Arn, Name, Description}. If at least customizationArn is provided,
-            //    the command selects this customization.
-            if (startUrl && region) {
-                await connectToEnterpriseSso(startUrl, region)
-            } else {
-                await getStartUrl()
-            }
-            if (customizationArn) {
-                const match = getPersistedCustomizations().find(c => c.arn == customizationArn)
-                const customization = {
-                    arn: customizationArn,
-                    name: customizationName ?? match?.name ?? 'unknown',
-                    description: customizationDescription ?? match?.description ?? 'unknown',
-                }
-                await selectCustomization(customization)
-            }
+    () => async (startUrl?: string, region?: string, customizationArn?: string, customizationNamePrefix?: string) => {
+        // This command supports two sets of arguments:
+        //  * startUrl and region. If both arguments are provided they will be used, otherwise
+        //    the command prompts for them interactively.
+        //  * customization{Arn, Name}. Selects a customization based on whether a match for given values are found.
+        if (startUrl && region) {
+            await connectToEnterpriseSso(startUrl, region)
+        } else {
+            await getStartUrl()
+        }
+
+        // No customization match information given, exit early.
+        if (!customizationArn && !customizationNamePrefix) {
+            return
         }
+
+        let match: Customization | undefined
+        // If given an ARN, assume a specific customization is desired and find an entry that matches it.
+        // Ignores the prefix logic. Backwards compatible with previous implementation.
+        if (customizationArn) {
+            match = getPersistedCustomizations().find(c => c.arn == customizationArn)
+        }
+        // Otherwise if only a prefix is given, find an entry that matches it.
+        else if (customizationNamePrefix) {
+            match = getPersistedCustomizations().find(c => c.name?.startsWith(customizationNamePrefix))
+        }
+
+        // If no match is found, nothing to do :)
+        if (!match) {
+            return
```

Comment:
should this throw a ToolkitError ? or at least log something.

Reply:
I think a log might be more appropriate here.
We might not want to throw since it's a request from outside of Toolkit, it can easily pass a value that isn't expected.

Reply:
> since it's a request from outside of Toolkit, it can easily pass a value that isn't expected.

If an external thing makes an invalid request to the Toolkit why would it expect the Toolkit to silently ignore that?